pkg/tools/serialize/serialize_test.go

```diff
@@ -138,3 +139,20 @@ func TestAsyncExecOneAtATime(t *testing.T) {
 		<-done
 	}
 }
+
+func BenchmarkExecutorAsync(b *testing.B) {
+	var exec serialize.Executor
+	var wg sync.WaitGroup
+	for i := 0; i < b.N; i++ {
+		wg.Add(1)
+		exec.AsyncExec(func() { wg.Done() })
```

Comment:
The benchmark might be more pure without using wg. To achieve this we can schedule b.N-1 executions inside the loop, and do the last execution after the loop using '<-exec.AsyncExec(func() {})' to ensure synchronization.